src/extensions/jana/JOmniFactoryGeneratorT.h

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023, Jefferson Science Associates, LLC.
+// Subject to the terms in the LICENSE file found in the top-level directory.
+// Created by Nathan Brei
+
+#pragma once
+
+#include <JANA/JFactorySet.h>
+#include <JANA/JFactoryGenerator.h>
+#include <vector>
+
+template<class FactoryT>
+class JOmniFactoryGeneratorT : public JFactoryGenerator {
+public:
+    using FactoryConfigType = typename FactoryT::ConfigType;
+
+private:
+    struct TypedWiring {
+        std::string m_tag;
+        std::vector<std::string> m_default_input_tags;
+        std::vector<std::string> m_default_output_tags;
+        FactoryConfigType m_default_cfg; /// Must be properly copyable!
+    };
+
+    struct UntypedWiring {
+        std::string m_tag;
+        std::vector<std::string> m_default_input_tags;
+        std::vector<std::string> m_default_output_tags;
+        std::map<std::string, std::string> m_config_params;
+    };
```

Comment:
How about
```suggestion
    struct StructWiring {
        std::string m_tag;
        std::vector<std::string> m_default_input_tags;
        std::vector<std::string> m_default_output_tags;
        FactoryConfigType m_default_cfg; /// Must be properly copyable!
    };
    
    struct TypedWiring {
        std::string m_tag;
        std::vector<std::string> m_default_input_tags;
        std::vector<std::string> m_default_output_tags;
        std::map<std::string, PropertyValueType> m_config_params;
    };

    struct UntypedWiring {
        std::string m_tag;
        std::vector<std::string> m_default_input_tags;
        std::vector<std::string> m_default_output_tags;
        std::map<std::string, std::string> m_config_params;
    };
```
with `using PropertyValueType = typename algorithms::PropertyValue;` from https://github.com/eic/algorithms/blob/main/core/include/algorithms/property.h#L33. That can add more type-safety for when desired.

Unfortunately no `FactoryT::PropertyValue`...

Reply:
Related to this: https://github.com/eic/EICrecon/pull/1154, where std::map<std::string, PropertyValueType> m_config_params is used since it allows setting defaults without parsing.

Reply:
Yeah I think that's great. Hopefully this will solve the problem I was facing where having an open universe of possible parameter types was preventing me from cleanly separating Omnifactories from Algorithms without resorting to Concepts.

Reply:
And what would be wrong with concepts anyway? ;-)
```cpp
    template < class T = FactoryConfigType >
        requires std::same_as<T, std::map<typename T::key_type, typename T::mapped_type, typename T::key_compare, typename T::allocator_type>>
    struct TypedWiring {
        std::string m_tag;
        std::vector<std::string> m_default_input_tags;
        std::vector<std::string> m_default_output_tags;
        std::map<std::string, typename FactoryConfigType::mapped_type> m_config_params;
    };
```

Reply:
But realistically, without any way to test this TypedWiring struct here, I think this probably better postponed until https://github.com/eic/EICrecon/pull/1154, after merging here and rebasing there.

So, resolving here.